app/internal/page/certificateprovider/enter_reference_code.go

```diff
@@ -0,0 +1,82 @@
+package certificateprovider
+
+import (
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/validation"
+
+	"github.com/ministryofjustice/opg-go-common/template"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/page"
+)
+
+type enterReferenceCodeData struct {
+	App    page.AppData
+	Errors validation.List
+	Form   *enterReferenceCodeForm
+}
+
+func EnterReferenceCode(tmpl template.Template, lpaStore LpaStore, dataStore page.DataStore) page.Handler {
+	return func(appData page.AppData, w http.ResponseWriter, r *http.Request) error {
+		data := enterReferenceCodeData{
+			App:  appData,
+			Form: &enterReferenceCodeForm{},
+		}
+
+		if r.Method == http.MethodPost {
+			data.Form = readEnterReferenceCodeForm(r)
+			data.Errors = data.Form.Validate()
+
+			if len(data.Errors) == 0 {
+				shareCode := data.Form.ReferenceCode
+
+				var v page.ShareCodeData
+				if err := dataStore.Get(r.Context(), "SHARECODE#"+shareCode, "#METADATA#"+shareCode, &v); err != nil {
+					return err
+				}
+
```

Comment:
If the sharecode can't be retrieved we should show a validation error because it is probably just a mistype. (I can't recall why I made dynamo.Client.Get return a `nil` error if the item couldn't be retrieved, but that is what it currently does)

Reply:
I did have incorrect codes at the back of my head but was just falling back on the error - hadn't realised `Get()` was defaulting to a nil error. Agree having a validation error would be nicer though - I'll take a look.